test/regression/issue/26515.test.ts

```
@@ -0,0 +1,94 @@
+
+    try {
+      // Port 1 is privileged and should fail immediately with connection refused
+      await fetch("http://localhost:1", {
+        method: "POST",
+        body: inputStream,
+        duplex: "half",
+      });
+    } catch (err: any) {
+      fetchError = err.message;
+    }
+  });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)